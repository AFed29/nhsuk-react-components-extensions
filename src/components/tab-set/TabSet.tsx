import classNames from 'classnames';
import React, { HTMLProps } from 'react';
import { Col, ColProps, Row } from 'react-flexbox-grid';

declare type TabProps = {
  isActive?: boolean;
  isDisabled?: boolean;
  isEmpty?: boolean;
  autosize?: boolean;
  xs?: string;
  sm?: string;
  md?: string;
  lg?: string;
  xl?: string;
};

const Tab: React.FC<HTMLProps<HTMLSpanElement> & ColProps & TabProps> = ({
  children,
  className,
  isActive,
  isDisabled,
  isEmpty,
  autosize,
  xs,
  sm,
  md,
  lg,
  xl,
  ...rest
}) => (
  <Col
    className={classNames(
      `nhsuk-tab-set__tab${
        // eslint-disable-next-line no-nested-ternary
        isDisabled ? '--disabled' : isActive ? '--active' : isEmpty ? '--empty' : ''
      }`,
      className,
    )}
    {...(autosize
      ? {}
      : {
        xs,
        sm,
        md,
        lg,
        xl,
      })}
  >
    {!isEmpty ? <span {...rest}>{children}</span> : null}
  </Col>
);

interface TabSet extends React.FC<HTMLProps<HTMLDivElement>> {
  Tab: React.FC<HTMLProps<HTMLSpanElement> & ColProps & TabProps>;
}

const TabSet: TabSet = ({ children, className, ...rest }) => (
  <div className={classNames('nhsuk-tab-set', className)} {...rest}>
    <Row>{children}</Row>
  </div>
);

TabSet.Tab = Tab;
export default TabSet;
