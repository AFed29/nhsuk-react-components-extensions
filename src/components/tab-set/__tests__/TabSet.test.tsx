/* eslint-disable jsx-a11y/label-has-associated-control */
import { mount, shallow } from 'enzyme';
import React from 'react';
import TabSet from '../TabSet';

describe('TabSet', () => {
  it('matches snapshot', () => {
    const component = shallow(
      <TabSet>
        <TabSet.Tab>Overview</TabSet.Tab>
        <TabSet.Tab isActive>Demographics</TabSet.Tab>
        <TabSet.Tab>Clinicals</TabSet.Tab>
        <TabSet.Tab>Documents</TabSet.Tab>
      </TabSet>,
    );
    expect(component).toMatchSnapshot();
    component.unmount();
  });

  it('calls onClick function', () => {
    const onClickFun = jest.fn();
    const wrapper = mount(
      <TabSet>
        <TabSet.Tab onClick={onClickFun} id="the-tab">
          Overview
        </TabSet.Tab>
      </TabSet>,
    );
    wrapper.find('span#the-tab').simulate('click');

    expect(onClickFun).toHaveBeenCalled();

    wrapper.unmount();
  });
});
