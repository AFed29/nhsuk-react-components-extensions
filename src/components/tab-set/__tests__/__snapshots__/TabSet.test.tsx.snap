// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabSet matches snapshot 1`] = `
<div
  className="nhsuk-tab-set"
>
  <Row>
    <Tab>
      Overview
    </Tab>
    <Tab
      isActive={true}
    >
      Demographics
    </Tab>
    <Tab>
      Clinicals
    </Tab>
    <Tab>
      Documents
    </Tab>
  </Row>
</div>
`;
